File: templates/project-scaffold/c/tests/test_core.c
/**
 * {{PROJECT_NAME}} — 核心算法测试
 *
 * 编译: make test
 * 运行: ./build/test_core
 */
#include <assert.h>
#include <stdio.h>
#include <string.h>
#include "../src/core.h"

static int tests_run = 0;
static int tests_passed = 0;

#define TEST(name) do { \
    printf("  [TEST] %s ... ", #name); \
    tests_run++; \
    name(); \
    tests_passed++; \
    printf("PASS\n"); \
} while(0)

/* ===== 测试用例 — 由 /ds:test 填充 ===== */

void test_placeholder(void) {
    assert(1 == 1);
}

// void test_basic_functionality(void) {
//     /* 基础功能测试 */
// }

// void test_edge_cases(void) {
//     /* 边界条件测试 */
// }

/* ===== 测试运行器 ===== */

int main(void) {
    printf("[OMD] Running tests...\n");

    TEST(test_placeholder);
    // TEST(test_basic_functionality);
    // TEST(test_edge_cases);

    printf("\n[OMD] Results: %d/%d passed\n", tests_passed, tests_run);
    return tests_passed == tests_run ? 0 : 1;
}

File: templates/project-scaffold/python/tests/test_core.py
"""{{PROJECT_NAME}} — 核心算法单元测试

运行: pytest tests/test_core.py -v
"""
import pytest
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent.parent / "src"))


class TestCoreAlgorithm:
    """核心算法测试套件 — 由 /ds:test 填充具体测试用例"""

    def test_placeholder(self):
        """占位测试 — 确认测试框架可运行"""
        assert True, "测试框架运行正常"

    # --- 以下由 /ds:implement 和 /ds:test 生成 ---

    # def test_basic_functionality(self):
    #     """基础功能测试"""
    #     pass

    # def test_edge_cases(self):
    #     """边界条件测试"""
    #     pass

    # def test_correctness(self):
    #     """正确性验证"""
    #     pass

    # def test_performance(self):
    #     """性能测试"""
    #     pass
